### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by JantuDeb**
# Wishlist
- Add and remove from wishlist
- Add to cart from wishlist
- Wishlist Item count in navbar
image

### CODE REVIEW

**Comment by hussiantabha** [comment]
One suggestion, On the cart page, keep the plus button in the right and minus on the left. Otherwise looks good

File: src/context/products/wishlist-reducer.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const ADD_TO_WISHLIST = "ADD_TO_WISHLIST";
+export const REMOVE_FROM_WISHLIST = "REMOVE_FROM_WISHLIST";
+
+export const wishlistReducer = (state, action) => {
+  const { type, payload } = action;
```
Comment (t007rushi) [inline src/context/products/wishlist-reducer.js:5]: liked the way you destructured code here for code reducibility

File: src/context/products/ProductContext.jsx [hunk 2]
```diff
@@ -25,7 +27,6 @@ const ProductProvider = ({ children }) => {
       console.log(error);
     }
   };
-
   const filterProducts = composeFilterFunc(
```
Comment (t007rushi) [inline src/context/products/ProductContext.jsx:30]: nice approach toward using compose functionality

**Review by t007rushi — APPROVED** [review]
clean code writing practices good to merge

File: src/context/products/ProductContext.jsx [hunk 3]
```diff
@@ -17,6 +18,7 @@ const ProductProvider = ({ children }) => {
     initialState
   );
   const [cartState, cartDispatch] = useReducer(cartReducer, []);
+  const [wishlistState, wishlistDispatch] = useReducer(wishlistReducer, []);
   const loadProducts = async () => {
     try {
       const { data } = await axiosInstance.get("/products");
```
Comment (sushil-mod) [inline src/context/products/ProductContext.jsx:24]: like how destructure direct data from response

**Review by sushil-mod — APPROVED** [review]
I like responsiveness and learn few things from your code . Clean code

**Pull request merged**